### ISSUE TITLE
double backwards on deform_conv2d not supporte

### ISSUE THREAD

**Issue by jiangxingxian** [pytorch/vision#9094]
### 🐛 Describe the bug

I encountered an error stating "operation not supported during the second backpropagation" when using deformable convolution and meta-learning. Can this issue be resolved?
class DeformConv2D_v2(nn.Module):
    def __init__(self, inc, outc, kernel_size=3, padding=1, stride=1):
        super().__init__()
        self.in_channels = inc
        self.out_channels = outc
        self.kernel_size = kernel_size
        self.padding = padding

        self.conv = nn.Conv2d(inc, outc, kernel_size, stride=stride, padding=padding) #原卷积
 
        self.conv_offset = nn.Conv2d(inc, 2*(kernel_size**2), kernel_size, stride=stride, padding=padding)
        init_offset = torch.Tensor(np.zeros([2*(kernel_size**2), inc, kernel_size, kernel_size]))
        self.conv_offset.weight = torch.nn.Parameter(init_offset) #初始化为0
 
        self.conv_mask = nn.Conv2d(inc, kernel_size**2, kernel_size, stride=stride, padding=padding)
        init_mask = torch.Tensor(np.zeros([kernel_size**2, inc, kernel_size, kernel_size])+np.array([0.5]))
        self.conv_mask.weight = torch.nn.Parameter(init_mask) #初始化为0.5
 
    def forward(self, x):
        offset = self.conv_offset(x)
        # print(x.shape,offset.shape, self.conv.weight.shape)
        mask = torch.sigmoid(self.conv_mask(x)) #保证在0到1之间
        out = torchvision.ops.deform_conv2d(input=x, offset=offset, 
                                            weight=self.conv.weight, 
                                             mask=mask, padding=(self.padding, self.padding))
        return out

### Versions

  File "/home/cxg3004_jbs/code/Distill_model/train_cd_sd_cb.py", line 821, in <module>
    train(config, model, train_data_loader, optimizer, scheduler, config.loss_list, snr, psnr, ssim, inner_objective, outer_objective,
  File "/home/cxg3004_jbs/code/Distill_model/train_cd_sd_cb.py", line 394, in train
    optimizer_target.meta_backward(state, model_source)
  File "/home/cxg3004_jbs/code/Distill_model/utils/meta_optimizers.py", line 113, in meta_backward
    X.backward()                                                    
  File "/home/cxg3004_jbs/.conda/envs/jbs2/lib/python3.8/site-packages/torch/_tensor.py", line 487, in backward
    torch.autograd.backward(
  File "/home/cxg3004_jbs/.conda/envs/jbs2/lib/python3.8/site-packages/torch/autograd/__init__.py", line 200, in backward
    Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
double backwards on deform_conv2d not supported

**Comment by AntoineSimoulin** [comment]
Hey @jiangxingxian thank you for posting, and apologies for the delayed response. Could you please provide a minimal reproducible example so that I can replicate the issue on my end? I don’t have access to your local files located at "/home/cxg3004_jbs/...", and I’m also unsure how the `DeformConv2D_v2` class is being used within the rest of your training loop. If you could share a minimal code snippet that clearly outlines the problem, it would greatly help me in debugging and providing you with effective support.
